company/okrs/2020_q1.md

```diff
@@ -0,0 +1,114 @@
+   1. VP Eng KR: Each team completes >80% of the work that they committed to at the beginning of the release cycle each iteration.
+
+1. **Distribution: Resolve issues for critical customer deals and expansions**
+1. **Distribution: Improve developer productivity**
+   1. Distribution KR: Lower build times to X
+   1. Distribution KR: Automate away release captain role
+   1. Distribution KR: Infrastructure for automated deploys and releases (no human in loop required by default)
+1. **Distribution: Build features that accelerate customers through pipeline**
+   1. Distribution KR: Shorten the time from "intent to try Sourcegraph" to "whole org is using Sourcegraph" (onboarding experience)
+   1. Distribution KR: Ensure new features (LSIF, GitLab Native Integration, Automation) are accessible and easily tryable by customers once they go GA
+   1. Distribution KR: Surface information to site admins and managers to encourage expansion/upsell
+
+1. **Code intelligence: LSIF is adopted by customers**
+   1. Code intelligence KR: At least three customers are using LSIF on their primary repositories
+   1. Code intelligence KR: At least one customer is using LSIF across all of their code in languages for which there exists an LSIF indexer
+
+1. **Code intelligence: Increase user confidence in code intelligence provided by LSIF**
+   1. Code intelligence KR: Code intelligence UI (tooltips, references panel, etc.) indicates whether or not results are precise
+   1. Code intelligence KR: Fall back to fuzzy if the file has been modified since the last LSIF upload
+   1. Code intelligence KR: Make use of LSIF data when the current line is not part of the diff since the last LSIF upload
+
+1. **Code intelligence: Make search better using precise code intel data**
+   1. Code intelligence KR: Implement search within dependencies and/or dependents of current repo
+   1. Code intelligence KR: Implement searching for defs/refs by symbol name
+
+1. **Web: Make users aware of advanced search features**
+   1. Web KR: Start tracking time to first search and search filters per query in pings
+   1. Web KR: Shorten time to first search
+   1. Web KR: Increase average number of search filters per query
+
+1. **Web: Automation is adopted by customers**
+   1. Web KR: At least four customers are using campaigns in production
+   1. Web KR: All campaign types have been used by customers in production
+
+1. **Core Services: Build features that accelerate customers through pipeline**
+   1. Core Services KR: Shorten the time from "intent to try Sourcegraph" to "whole org is using Sourcegraph" (onboarding experience)
+   1. Core Services KR: Ensure new features (Automation, ACLs, etc) are accessible and easily tryable by customers once they go GA
+
+1. **Core Services: Search has feature parity with OpenGrok**
+   1. Core Services KR: Implement AND query support
+   1. Core Services KR: Implement OR query support
+   1. Core Services KR: Implement NOT query support
```

Comment:
I'm not sure it makes sense to break this down by operator. We should think about the expression language as one unit. The part that we'll have to expand is where and how this language applies (i.e., implementation that turns expressions into search functionality. In what order, if any, should we support file contents, file paths, repos, etc? AFAIK OpenGrok basically only works with expressions dealing with file content (and maybe file paths? not sure). More considerations to come in RFC...

Reply:
Please propose the KRs that you think make sense. I do think that the O is a good one to aspire to though.

Reply:
KRs, two ways to phrase this, roughly:
- Users can search for file contents using `AND`/`OR`/`NOT` operators 

or

- Sourcegraph supports a more expressive way to search file contents using `AND`/`OR`/`NOT` operators 

I'm wording with a restriction on "file contents" as the initial feature set. We want to expand this out to repos/file paths, but there are some "unknown unknowns" at this point and I don't know how much complexity that will import So: file contents is a good first start. 

Note that implementing AND/OR/NOT functionality will implicitly allow a 'bag of words' style search (so not only more expressive, but also a new search kind). So a direct result of this is a follow up KR that we support bag of words search on file contents.